suite-common/connect-popup/src/methodHooks/ethereumSignTransaction.ts

```diff
@@ -0,0 +1,111 @@
+            );
+            if (!selectedAccount) {
+                // Create a new placeholder account
+                if (!device?.state?.staticSessionId) {
+                    throw new Error('Device state is missing');
+                }
+                await dispatch(
+                    accountsActions.createAccount({
+                        deviceState: device.state.staticSessionId,
+                        discoveryItem: {
+                            index: 0,
+                            path,
+                            accountType: 'imported',
+                            networkType: network.networkType,
+                            coin: network.symbol,
+                        },
+                        accountInfo: {
+                            path,
+                            descriptor: '',
+                            balance: '',
+                            availableBalance: '',
+                            empty: false,
+                            history: { total: 0, unconfirmed: 0 },
+                        },
+                        imported: true,
+                        visible: false,
+                        accountLabel: network.name,
+                    }),
+                );
+                selectedAccount = selectAccounts(getState()).find(
+                    account =>
+                        path === account.path &&
+                        account.symbol === network.symbol &&
+                        account.imported === true,
+                );
+            }
+            if (!selectedAccount) {
+                throw new Error('Account could not be created');
+            }
+            dispatch(
+                connectPopupActions.setSelectedAccountKey({
+                    selectedAccountKey: selectedAccount.key,
+                }),
+            );
+            dispatch(
+                sendFormActions.storePrecomposedTransaction({
+                    formState: {
+                        // Can be left empty, not used in tx review modal
+                        outputs: [],
+                        feeLimit: '',
+                        feePerUnit: '',
+                        selectedUtxos: [],
+                        isCoinControlEnabled: false,
+                        hasCoinControlBeenOpened: false,
+                        options: ['ethereumNonce', 'ethereumData'],
+                        selectedFee: 'custom',
+                        ethereumDataAscii: '',
+                        ethereumDataHex: typedPayload.transaction.data?.replace(/^0x/, ''),
+                        ethereumNonce: typedPayload.transaction.nonce,
+                    },
+                    precomposedTransaction: txSigningPrecomposed,
+                }),
+            );
+        }
+    } catch (error) {
+        // If an error occurs it's not a problem, we just fall back to generic UI
+        console.error(`Error in Connect Popup ${method} hook:`, error);
```

Comment:
soo nothing here is "business critical" and the call flow can continue?

Reply:
Yeah, it will just show the generic "Confirm on device" popup